public/app/features/transformers/extractFields/fieldExtractor.test.ts

```diff
@@ -14,6 +14,37 @@ describe('Extract fields from text', () => {
     `);
   });
 
+  it('Test key-values with single/double quotes', async () => {
+    const extractor = fieldExtractors.get(FieldExtractorID.KeyValues);
+    const out = extractor.parse('a="1",   "b"=\'2\',c=3  x:y ;\r\nz="d and 4"');
+    expect(out).toMatchInlineSnapshot(`
+      Object {
+        "a": "1",
+        "b": "2",
+        "c": "3",
+        "x": "y",
+        "z": "d and 4",
+      }
+    `);
+  });
+
+  it('Test key-values with nested single/double quotes', async () => {
+    const extractor = fieldExtractors.get(FieldExtractorID.KeyValues);
+    const out = extractor.parse(
+      `a="1",   "b"=\'2\',c=3  x:y ;\r\nz="dbl_quotes="Double Quotes" sgl_quotes='Single Quotes'"`
```

Comment:
this input seems wrong --- I would expect it to parse into:
`nz=dbl_quotes` 
and then giberish

Reply:
Have updated the test for this with the correct escaping